gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java

```diff
@@ -16,15 +16,469 @@
+      return answer == null ? null : DiskType.fromPb(answer);
+    } catch (RetryHelper.RetryHelperException e) {
+      throw ComputeException.translateAndThrow(e);
+    }
+  }
+
+  @Override
+  public DiskType getDiskType(String zone, String diskType, DiskTypeOption... options)
+      throws ComputeException {
+    return getDiskType(DiskTypeId.of(zone, diskType), options);
+  }
+
+  @Override
+  public Page<DiskType> listDiskTypes(String zone, DiskTypeListOption... options)
+      throws ComputeException {
+    return listDiskTypes(zone, options(), optionMap(options));
+  }
+
+  private static Page<DiskType> listDiskTypes(final String zone,
+      final ComputeOptions serviceOptions, final Map<ComputeRpc.Option, ?> optionsMap) {
+    try {
+      ComputeRpc.Tuple<String, Iterable<com.google.api.services.compute.model.DiskType>> result =
+          runWithRetries(new Callable<ComputeRpc.Tuple<String,
+              Iterable<com.google.api.services.compute.model.DiskType>>>() {
+            @Override
+            public ComputeRpc.Tuple<String,
+                Iterable<com.google.api.services.compute.model.DiskType>> call() {
+              return serviceOptions.rpc().listDiskTypes(zone, optionsMap);
+            }
+          }, serviceOptions.retryParams(), EXCEPTION_HANDLER);
+      String cursor = result.x();
+      Iterable<DiskType> diskTypes = Iterables.transform(
+          result.y() == null ? ImmutableList.<com.google.api.services.compute.model.DiskType>of()
+              : result.y(),
+          new Function<com.google.api.services.compute.model.DiskType, DiskType>() {
+            @Override
+            public DiskType apply(com.google.api.services.compute.model.DiskType diskType) {
+              return DiskType.fromPb(diskType);
+            }
+          });
+      return new PageImpl<>(new DiskTypePageFetcher(zone, serviceOptions, cursor, optionsMap),
+          cursor, diskTypes);
+    } catch (RetryHelper.RetryHelperException e) {
+      throw ComputeException.translateAndThrow(e);
+    }
+  }
+
+  @Override
+  public Page<DiskType> listDiskTypes(DiskTypeListOption... options) throws ComputeException {
+    return listDiskTypes(options(), optionMap(options));
+  }
+
+  private static Page<DiskType> listDiskTypes(final ComputeOptions serviceOptions,
+      final Map<ComputeRpc.Option, ?> optionsMap) {
+    try {
+      ComputeRpc.Tuple<String, Iterable<com.google.api.services.compute.model.DiskType>> result =
+          runWithRetries(new Callable<ComputeRpc.Tuple<String,
+              Iterable<com.google.api.services.compute.model.DiskType>>>() {
+            @Override
+            public ComputeRpc.Tuple<String,
+                Iterable<com.google.api.services.compute.model.DiskType>> call() {
+              return serviceOptions.rpc().listDiskTypes(optionsMap);
+            }
+          }, serviceOptions.retryParams(), EXCEPTION_HANDLER);
+      String cursor = result.x();
```

Comment:
Ok, Fine as is, but I don't think we need to consider that as a recommended way or a pattern we should follow (nor should we go and change other places).